macOS: Bump minimum required runtime version and prepare for building with upstream LLVM

theuni opened the pull request:
This (I believe) resolves the last of the blockers for [switching us away from cctools and instead using out-of-the-box llvm and lld](https://github.com/bitcoin/bitcoin/pull/21778) for building Darwin binaries.

For now, we continue building with a pre-packaged llvm and cctools, but after this PR the clang+lld combo should just work for anyone trying it. Additionally after this PR, the new runtime `fixup_chains` behavior will be in-use, as ld64 uses it as well.

The commits may seem unrelated, so in detail:

lld (llvm's linker) has been a work-in-progress for Darwin for years. Recently though, it has gained nearly all of the features we require. The last missing feature from ld64, `-Wl,-bind_at_load`, is not implemented in lld; as far as I can tell [lazy loading has conceptually been replaced by fixup chains](https://www.emergetools.com/blog/posts/iOS15LaunchTime).

So that means we don't need ld64's `bind_at_load` as long as lld can handle `-Wl,-fixup_chains` (which it can). I've added it to our configure as a linker option mostly so that we can see it in the logs; it's default-on as long as the minimum version is >11.0.

About that: the runtime functionality required for `-Wl,-fixup_chains` [requires macOS >=11.0](https://github.com/llvm/llvm-project/blob/release/16.x/lld/MachO/Driver.cpp#L1021). Hence the commit that bumps the minimum version. Our current min runtime of `10.15` has been unsupported since September 2022, so I don't expect this bump to be controversial.

Lastly, with the minimum runtime version bumped to 11.0, our current version of pre-compiled clang we use for macOS is too old to understand `-mmacosx-version-min=11.0` because it expects `=10.x`. So I've made the smallest possible bump (from 10.0.1 to 11.1.0) to a version that understands. This bump is arbitrary and unfortunate, but likely to be short-lived as we may end up replacing it with llvm+lld for v26 anyway. I've held off on bumping the SDK as I think that makes sense to do as part of the lld switch instead.

theuni committed:
depends: Bump MacOS minimum runtime requirement to 11.0

This is necessary as the new fixup_chains linker behavior is only valid
when the runtime target is >=11.0.

theuni committed:
build: explicitly opt-in to new fixup_chains functionality for darwin

This replaces (but does not collide with) the previous bind_on_load. There
is technically no need to opt-in to this functionality as long as >= MacOS 11.0
is being targetted, but it will be helpful to see in the logs.

theuni committed:
depends: bump darwin clang to 11.1

Unfortunately clang 10 does not understand "-mmacosx-version-min=11.0",
as it expects to see only 10.x.

Bump minimally to 11.1 to fix that problem. This will likely be our last
binary toolchain bump, as it will soon be replaced with usage of upstream
vanilla llvm.

theuni commented:
Ping @fanquake and @hebasto . There's not much happening here technically, but a few big-ish conceptual changes.

Also, we may want to go one step further in configure and do an actual compile check to verify that the minimum version is set to >=11.0 as we now have that as an assumption.

fanquake commented:
Concept ACK - note that this also needs a change to `clang-toolchain-11` in Guix, which should be available without a time-machine change.

hebasto commented:
Concept ACK.

fanquake commented:
> check to verify that the minimum version is set to >=11.0 as we now have that as an assumption.

Note that you'll also need to update our minimum version check in:
https://github.com/bitcoin/bitcoin/blob/904631e0fc00ac9c8a03d1ce226d071bf88c00db/contrib/devtools/symbol-check.py#L235

theuni commented:
> Note that you'll also need to update our minimum version check in:
> 
> https://github.com/bitcoin/bitcoin/blob/904631e0fc00ac9c8a03d1ce226d071bf88c00db/contrib/devtools/symbol-check.py#L235

Thanks. Done. I used #22993 as a template for bumping, we'll see if c-i is happy.

theuni commented:
Pushed a test for `fixup_chains`, but I don't think it ever ran. @fanquake What sets `RUN_SECURITY_TESTS`?

fanquake commented:
>  @fanquake What sets RUN_SECURITY_TESTS?

Nothing. symbol/security checks are no-longer run in the CI, because the CI environment is semi-regularly changing, and the binaries produced don't represent release binaries, meaning there are some tests that will never pass. `RUN_SECURITY_TESTS` is leftover from when we used run (some) tests; I've opened a PR to remove it (#27683). The current way to run the security/symbol checks is to perform a Guix build.

fanquake commented:
The CI failure is unrelated, and will be fixed post-rebase. Guix currently fails to build. i.e:
```bash
time HOSTS="x86_64-apple-darwin arm64-apple-darwin" ./contrib/guix/guix-build
<snip>
Extracting libevent...
/home/ubuntu/sources/libevent-2.1.12-stable.tar.gz: OK
Preprocessing libevent...
Configuring libevent...
checking for a BSD-compatible install... /home/ubuntu/.guix-profile/bin/install -c
checking whether build environment is sane... yes
checking for x86_64-apple-darwin-strip... x86_64-apple-darwin-strip
checking for a thread-safe mkdir -p... /home/ubuntu/.guix-profile/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking whether make supports the include directive... yes (GNU style)
checking for x86_64-apple-darwin-gcc... env -u C_INCLUDE_PATH -u CPLUS_INCLUDE_PATH -u OBJC_INCLUDE_PATH -u OBJCPLUS_INCLUDE_PATH -u CPATH -u LIBRARY_PATH /home/ubuntu/.guix-profile/bin/clang --target=x86_64-apple-darwin -mmacosx-version-min=11.0 -B/bitcoin/depends/x86_64-apple-darwin/native/bin -mlinker-version=609 -isysroot/home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers -Xclang -internal-externc-isystem -Xclang /gnu/store/2vd3s80pipf4dffzrf4niphl2mdhrjx1-clang-11.0.0/lib/clang/11.0.0/include -Xclang -internal-externc-isystem -Xclang /home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers/usr/include
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether env -u C_INCLUDE_PATH -u CPLUS_INCLUDE_PATH -u OBJC_INCLUDE_PATH -u OBJCPLUS_INCLUDE_PATH -u CPATH -u LIBRARY_PATH /home/ubuntu/.guix-profile/bin/clang --target=x86_64-apple-darwin -mmacosx-version-min=11.0 -B/bitcoin/depends/x86_64-apple-darwin/native/bin -mlinker-version=609 -isysroot/home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers -Xclang -internal-externc-isystem -Xclang /gnu/store/2vd3s80pipf4dffzrf4niphl2mdhrjx1-clang-11.0.0/lib/clang/11.0.0/include -Xclang -internal-externc-isystem -Xclang /home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers/usr/include accepts -g... yes
checking for env -u C_INCLUDE_PATH -u CPLUS_INCLUDE_PATH -u OBJC_INCLUDE_PATH -u OBJCPLUS_INCLUDE_PATH -u CPATH -u LIBRARY_PATH /home/ubuntu/.guix-profile/bin/clang --target=x86_64-apple-darwin -mmacosx-version-min=11.0 -B/bitcoin/depends/x86_64-apple-darwin/native/bin -mlinker-version=609 -isysroot/home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers -Xclang -internal-externc-isystem -Xclang /gnu/store/2vd3s80pipf4dffzrf4niphl2mdhrjx1-clang-11.0.0/lib/clang/11.0.0/include -Xclang -internal-externc-isystem -Xclang /home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers/usr/include option to accept ISO C89... none needed
checking whether env -u C_INCLUDE_PATH -u CPLUS_INCLUDE_PATH -u OBJC_INCLUDE_PATH -u OBJCPLUS_INCLUDE_PATH -u CPATH -u LIBRARY_PATH /home/ubuntu/.guix-profile/bin/clang --target=x86_64-apple-darwin -mmacosx-version-min=11.0 -B/bitcoin/depends/x86_64-apple-darwin/native/bin -mlinker-version=609 -isysroot/home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers -Xclang -internal-externc-isystem -Xclang /gnu/store/2vd3s80pipf4dffzrf4niphl2mdhrjx1-clang-11.0.0/lib/clang/11.0.0/include -Xclang -internal-externc-isystem -Xclang /home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers/usr/include understands -c and -o together... yes
checking dependency style of env -u C_INCLUDE_PATH -u CPLUS_INCLUDE_PATH -u OBJC_INCLUDE_PATH -u OBJCPLUS_INCLUDE_PATH -u CPATH -u LIBRARY_PATH /home/ubuntu/.guix-profile/bin/clang --target=x86_64-apple-darwin -mmacosx-version-min=11.0 -B/bitcoin/depends/x86_64-apple-darwin/native/bin -mlinker-version=609 -isysroot/home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers -Xclang -internal-externc-isystem -Xclang /gnu/store/2vd3s80pipf4dffzrf4niphl2mdhrjx1-clang-11.0.0/lib/clang/11.0.0/include -Xclang -internal-externc-isystem -Xclang /home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers/usr/include... none
checking how to run the C preprocessor... /lib/cpp
configure: error: in `/bitcoin/depends/work/build/x86_64-apple-darwin/libevent/2.1.12-stable-3b53fa1ad55':
configure: error: C preprocessor "/lib/cpp" fails sanity check
See `config.log' for more details
make: *** [funcs.mk:292: /bitcoin/depends/x86_64-apple-darwin/.libevent_stamp_configured] Error 1
make: Leaving directory '/bitcoin/depends'
```

config.log https://gist.github.com/fanquake/e0e1cc9f2976eb745dbb92a69c350994:
```bash
configure:4403: checking how to run the C preprocessor
configure:4434: env -u C_INCLUDE_PATH -u CPLUS_INCLUDE_PATH -u OBJC_INCLUDE_PATH -u OBJCPLUS_INCLUDE_PATH -u CPATH -u LIBRARY_PATH /home/ubuntu/.guix-profile/bin/clang --target=x86_64-apple-darwin -mmacosx-version-min=11.0 -B/bitcoin/depends/x86_64-apple-darwin/native/bin -mlinker-version=609 -isysroot/home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers -Xclang -internal-externc-isystem -Xclang /gnu/store/2vd3s80pipf4dffzrf4niphl2mdhrjx1-clang-11.0.0/lib/clang/11.0.0/include -Xclang -internal-externc-isystem -Xclang /home/ubuntu/SDKs/Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers/usr/include -E   -I/bitcoin/depends/x86_64-apple-darwin/include -D_FORTIFY_SOURCE=3 conftest.c
In file included from conftest.c:13:
In file included from /gnu/store/2vd3s80pipf4dffzrf4niphl2mdhrjx1-clang-11.0.0/lib/clang/11.0.0/include/limits.h:21:
In file included from /gnu/store/5h2w4qi9hk1qzzgi1w83220ydslinr4s-glibc-2.33/include/limits.h:26:
In file included from /gnu/store/5h2w4qi9hk1qzzgi1w83220ydslinr4s-glibc-2.33/include/bits/libc-header-start.h:33:
/gnu/store/5h2w4qi9hk1qzzgi1w83220ydslinr4s-glibc-2.33/include/features.h:397:4: warning: _FORTIFY_SOURCE requires compiling with optimization (-O) [-W#warnings]
#  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
   ^
In file included from conftest.c:13:
In file included from /gnu/store/2vd3s80pipf4dffzrf4niphl2mdhrjx1-clang-11.0.0/lib/clang/11.0.0/include/limits.h:21:
In file included from /gnu/store/5h2w4qi9hk1qzzgi1w83220ydslinr4s-glibc-2.33/include/limits.h:195:
In file included from /gnu/store/5h2w4qi9hk1qzzgi1w83220ydslinr4s-glibc-2.33/include/bits/posix1_lim.h:161:
/gnu/store/5h2w4qi9hk1qzzgi1w83220ydslinr4s-glibc-2.33/include/bits/local_lim.h:38:10: fatal error: 'linux/limits.h' file not found
#include <linux/limits.h>
         ^~~~~~~~~~~~~~~~
1 warning and 1 error generated.
```

theuni commented:
Converting to draft while I investigate the clang issue.

theuni commented:
It took several days to track down, but... I believe the issue with clang >=11 comes from: https://github.com/llvm/llvm-project/commit/6fa3894c4e771c773712b1ae777f78c1c922a908 . From my local tests I'm able to revert that and get back to expected results.

So I guess we'll need to patch it out of guix.

fanquake commented on depends/packages/native_clang.mk:
```diff
@@ -1,12 +1,12 @@
 $(package)_download_path=https://github.com/llvm/llvm-project/releases/download/llvmorg-$($(package)_version)
 ifneq (,$(findstring aarch64,$(BUILD)))
 $(package)_file_name=clang+llvm-$($(package)_version)-aarch64-linux-gnu.tar.xz
 $(package)_sha256_hash=90dc69a4758ca15cd0ffa45d07fbf5bf4309d47d2c7745a9f0735ecffde9c31f
```
```suggestion
$(package)_sha256_hash=18df38247af3fba0e0e2991fb00d7e3cf3560b4d3509233a14af699ef0039e1c
```

theuni commented:
> So I guess we'll need to patch it out of guix.

Suggestions for how to go about this are welcome. It's not at all straightforward to me how to patch out.

The llvm toolchain bumps are currently blocked on this :(

theuni commented:
Added a custom forked guix repo containing the patch we need for llvm's build.

I *believe* I've fixed authorization and it should just work(tm).

My local guix build works now, but I've reworked things so many times it would be helpful to get a confirmation.

Going to mark as ready for review since guix builds might actually work now.

theuni commented:
See relevant #25098. If we go forward with this we'll want the guix fork as part of our github org rather than my personal repo.

fanquake commented:
> I believe I've fixed authorization and it should just work(tm).
> it would be helpful to get a confirmation.

Guix builds are working. However the security checks don't pass, as `has_dyld_chained_fixups` seems to only be available with LIEF 0.13.0, and we are currently using `0.12.3`.

theuni commented:
> Guix builds are working.

Thanks for the build.

> However the security checks don't pass, as `has_dyld_chained_fixups` seems to only be available with LIEF 0.13.0, and we are currently using `0.12.3`.

Sigh. It just never ends. Converting back to draft as I don't intend to look into this immediately.

theuni commented:
After discussing offline with @fanquake, it doesn't make sense to let a single test keep us from upgrading our entire darwin toolchain infrastructure. I've removed the lief check for now until we can bump to >= 0.13.0.

fanquake committed:
contrib: remove macOS lazy_bind check

In future, this will be replaced by a check for fixup_chains usage.

fanquake commented:
Note that you'l have to drop the `lazy_bind` check (we'll follow up with a `fixup_chains` test later). I've got a branch here with that change, which Guix builds successfully: https://github.com/fanquake/bitcoin/tree/27676_minus_lazy_bind.

Guix Build:
```bash
2da11646ff030cbd9207027e5af33ec77e8dc30fa4811fb47e6c2201da9b54e5  guix-build-36d66d7df78e/output/arm64-apple-darwin/SHA256SUMS.part
e1c5a81e97a942ace7c57fd7c12b8f45287939dd193418c435ac806d28d76b28  guix-build-36d66d7df78e/output/arm64-apple-darwin/bitcoin-36d66d7df78e-arm64-apple-darwin-unsigned.dmg
c357cc2d55719ea7b4670242c9567aa30f6ba535593fcd7b00669a4a4bc29af5  guix-build-36d66d7df78e/output/arm64-apple-darwin/bitcoin-36d66d7df78e-arm64-apple-darwin-unsigned.tar.gz
be944333369b53aaab5fab6247e30a36a78314e6489538d2de0ab6b1961cb5ed  guix-build-36d66d7df78e/output/arm64-apple-darwin/bitcoin-36d66d7df78e-arm64-apple-darwin.tar.gz
7c3ce193da173af3acd0650fb10f684bc4ce22b5134cefe186c1206eddf0ca5c  guix-build-36d66d7df78e/output/dist-archive/bitcoin-36d66d7df78e.tar.gz
04b969c7af2ac9869e77c94d00db58a69916c9b9727b716562399de308138af5  guix-build-36d66d7df78e/output/x86_64-apple-darwin/SHA256SUMS.part
d592018b63c83b7b1e7532648b14e20209d326c5821ea3d22e9c6cd5b7640cbf  guix-build-36d66d7df78e/output/x86_64-apple-darwin/bitcoin-36d66d7df78e-x86_64-apple-darwin-unsigned.dmg
b9992629d71e840c90a6dd9ddcb3d19f95dabcc8e429c0c60b3b63a46b590ffe  guix-build-36d66d7df78e/output/x86_64-apple-darwin/bitcoin-36d66d7df78e-x86_64-apple-darwin-unsigned.tar.gz
c0a2386ee8f90b795b4d8ee27fa56e70c3d33db1a42c47fdc116c0a2f7d9db0a  guix-build-36d66d7df78e/output/x86_64-apple-darwin/bitcoin-36d66d7df78e-x86_64-apple-darwin.tar.gz
```

theuni commented:
> Note that you'l have to drop the `lazy_bind` check (we'll follow up with a `fixup_chains` test later). I've got a branch here with that change, which Guix builds successfully: https://github.com/fanquake/bitcoin/tree/27676_minus_lazy_bind.

Whoops, yes, thanks. Patch taken.

So this is now 100% working, just need to decide out what to do about guix.

hebasto commented:
> just need to decide out what to do about guix.

What options are currently being considered. Is an upstream patch still possible?

theuni commented:
> > just need to decide out what to do about guix.
> 
> What options are currently being considered. Is an upstream patch still possible?

Yes, I'm still working on testing this. It changes how the llvm build works and I'd like to be more confident in the change before proposing it.

theuni commented:
> > just need to decide out what to do about guix.
> 
> What options are currently being considered. Is an upstream patch still possible?

See #27798 for an alternative that is imo better in every way :)

theuni commented:
Rebased on top of #27798 rather than messing with guix.

fanquake commented:
Guix Build:
```bash
820344e9b2e233b970db497533db86462c6d435c9533c817ac8ce1ca2cccde1b  guix-build-a97f8f464b76/output/aarch64-linux-gnu/SHA256SUMS.part
d0d49c0d65d61fc067570a29bfb75dad0d79279308bc7e8d82bf1c41d71b54f2  guix-build-a97f8f464b76/output/aarch64-linux-gnu/bitcoin-a97f8f464b76-aarch64-linux-gnu-debug.tar.gz
33aacd5eb2073f2c0df4b8aad93febc0fe0e23899032520765e7f2f9443aa7ba  guix-build-a97f8f464b76/output/aarch64-linux-gnu/bitcoin-a97f8f464b76-aarch64-linux-gnu.tar.gz
ff9b706a194011a9ba1bcefb94503663baefd91712b32c45c8fbcd4b2703ea98  guix-build-a97f8f464b76/output/arm-linux-gnueabihf/SHA256SUMS.part
1691130319bc77d5f5bcbe311db9f20f595aa69158aaa74e69d37799a341ceb9  guix-build-a97f8f464b76/output/arm-linux-gnueabihf/bitcoin-a97f8f464b76-arm-linux-gnueabihf-debug.tar.gz
89ce50db33e15c41dfc91acfdc58c84bf2ab881894aa2de1011780ac26da3dcd  guix-build-a97f8f464b76/output/arm-linux-gnueabihf/bitcoin-a97f8f464b76-arm-linux-gnueabihf.tar.gz
298b52381fa6d7fdd62328f0db1c680c188fdfb14623099580f72f9aef5ccedc  guix-build-a97f8f464b76/output/arm64-apple-darwin/SHA256SUMS.part
9139b7fd5771aa0cf03680e358b2498eedda7bea4be92d71d8c852e0571e648b  guix-build-a97f8f464b76/output/arm64-apple-darwin/bitcoin-a97f8f464b76-arm64-apple-darwin-unsigned.dmg
4a6d23f2b4ebf3d5d865350406b9eb3b062846fbed35b4b9986cf0f0940179ca  guix-build-a97f8f464b76/output/arm64-apple-darwin/bitcoin-a97f8f464b76-arm64-apple-darwin-unsigned.tar.gz
fb9d6dc71cb24da81dd75a9a0cd25a32cc9517a3e84503c6a58a46d373b15af3  guix-build-a97f8f464b76/output/arm64-apple-darwin/bitcoin-a97f8f464b76-arm64-apple-darwin.tar.gz
def8eccf94c1b18824ea49da2c7e24256be0d72993d9ff9b69957754eb76f7c5  guix-build-a97f8f464b76/output/dist-archive/bitcoin-a97f8f464b76.tar.gz
7854007dc0dc6ab77eff32abed34e28f43b1639d1476f51de1fad702481acac4  guix-build-a97f8f464b76/output/powerpc64-linux-gnu/SHA256SUMS.part
753ce3cb5c5ac5c5f40a94e7e87d337215dd5564ec0fffe167d8000c6523ad39  guix-build-a97f8f464b76/output/powerpc64-linux-gnu/bitcoin-a97f8f464b76-powerpc64-linux-gnu-debug.tar.gz
4d0d2378e719a0119941b25f74f4058204011376289246b3d4d0649b7a169584  guix-build-a97f8f464b76/output/powerpc64-linux-gnu/bitcoin-a97f8f464b76-powerpc64-linux-gnu.tar.gz
392ee675bff2420b328b23bc5370b8a0b1ae20d52e4d7f2229f60a663770059d  guix-build-a97f8f464b76/output/powerpc64le-linux-gnu/SHA256SUMS.part
6011af479c016746ccda52c3f563c673a29a5a5861bb2da0466df0b91e72a1a9  guix-build-a97f8f464b76/output/powerpc64le-linux-gnu/bitcoin-a97f8f464b76-powerpc64le-linux-gnu-debug.tar.gz
f2153f79b102a1dd334b9ebc3693a6fad3329098604a3b9ed2dfbab92a8ead78  guix-build-a97f8f464b76/output/powerpc64le-linux-gnu/bitcoin-a97f8f464b76-powerpc64le-linux-gnu.tar.gz
92982dd466266d9c3d2b0fd2b8aed440a7d63b101bb1510c8449bbac90cb133d  guix-build-a97f8f464b76/output/riscv64-linux-gnu/SHA256SUMS.part
687538ef1f655af370607becbf62b486a8e15c1014d1e271118652cd56f349de  guix-build-a97f8f464b76/output/riscv64-linux-gnu/bitcoin-a97f8f464b76-riscv64-linux-gnu-debug.tar.gz
388dc1f394b676c13abb854c3723690187ada5ca51c34be7d09658f9d1f89785  guix-build-a97f8f464b76/output/riscv64-linux-gnu/bitcoin-a97f8f464b76-riscv64-linux-gnu.tar.gz
92e480b0317feedceeadd7d843a17ec862d37941d1b37ba0fe5b78c69419f4cb  guix-build-a97f8f464b76/output/x86_64-apple-darwin/SHA256SUMS.part
63aa9c2e562511da27252574681b701fc41b138461ae52a38fc953c22137ea49  guix-build-a97f8f464b76/output/x86_64-apple-darwin/bitcoin-a97f8f464b76-x86_64-apple-darwin-unsigned.dmg
0b6c47109e241a9305b14341e714453a8a752ab9fb086879ed8d39f608952b70  guix-build-a97f8f464b76/output/x86_64-apple-darwin/bitcoin-a97f8f464b76-x86_64-apple-darwin-unsigned.tar.gz
3e7c2491a87a200adf753cfb28e1daa7f7fffe941dfade579be66dd6cba54c1e  guix-build-a97f8f464b76/output/x86_64-apple-darwin/bitcoin-a97f8f464b76-x86_64-apple-darwin.tar.gz
2d1320eb57f6c7a12d3cbadf3e9d2019fc5b284835c8be290ed1a2752379c9ab  guix-build-a97f8f464b76/output/x86_64-linux-gnu/SHA256SUMS.part
2c0717fb67a89956b4d8082696b24c69b45b1b3248c2b890f06f64d15d64571b  guix-build-a97f8f464b76/output/x86_64-linux-gnu/bitcoin-a97f8f464b76-x86_64-linux-gnu-debug.tar.gz
9dbe6b47f2b34fa4b78f6e068032a8ecdc5cb76739d57d26fb96a4788948929f  guix-build-a97f8f464b76/output/x86_64-linux-gnu/bitcoin-a97f8f464b76-x86_64-linux-gnu.tar.gz
49a1c23783843d3cda29c994d6f0a10540b84ccf9ddac60247a09fafcbf2881e  guix-build-a97f8f464b76/output/x86_64-w64-mingw32/SHA256SUMS.part
7c04f54f337a2ac963fd6ccac8abf6bd3130519be845ac677717bc4287895bcd  guix-build-a97f8f464b76/output/x86_64-w64-mingw32/bitcoin-a97f8f464b76-win64-debug.zip
24a66d2b40d19a6741b9fa796dbc1ca2b3c6d01fd09e6f3bf93031a5da1777b9  guix-build-a97f8f464b76/output/x86_64-w64-mingw32/bitcoin-a97f8f464b76-win64-setup-unsigned.exe
b2490a696818c807c4cbc755381ccd01fc85ccb194dc6dfab9c6c0399cd7d448  guix-build-a97f8f464b76/output/x86_64-w64-mingw32/bitcoin-a97f8f464b76-win64-unsigned.tar.gz
af72801c7972b8f9594232eb514d2d837893a9aa42c1841ef2bb0fe334bf9f8c  guix-build-a97f8f464b76/output/x86_64-w64-mingw32/bitcoin-a97f8f464b76-win64.zip
```

fanquake commented:
Want to rebase and undraft this now that #27798 is in?

fanquake commented:
Guix Build:
```bash
109ed45706b8cbdc897b6c51efa0247f8e8f1e6bb35c57e69f30093b62189a51  guix-build-3df60704661c/output/aarch64-linux-gnu/SHA256SUMS.part
4e28463f1c074c7c121abd5fbad89aa2269c5978829ccd4898d33a21eedfeb4f  guix-build-3df60704661c/output/aarch64-linux-gnu/bitcoin-3df60704661c-aarch64-linux-gnu-debug.tar.gz
f20d074f0529202274a40f0530f79f601f5c575f1cd0addc099418719982485e  guix-build-3df60704661c/output/aarch64-linux-gnu/bitcoin-3df60704661c-aarch64-linux-gnu.tar.gz
414d7c4bf531a8d8589c15c9636ec0b092b28cba359b7c5ff46cd5bc0a768ac6  guix-build-3df60704661c/output/arm-linux-gnueabihf/SHA256SUMS.part
97e5cc1caf38e9e6cb4efde9be3a163c7373a1bdf1f7bdfbaa5fa7c9dd22a87c  guix-build-3df60704661c/output/arm-linux-gnueabihf/bitcoin-3df60704661c-arm-linux-gnueabihf-debug.tar.gz
cc174cb93c1b13b0efd5e002741ee6a13fcf1388f766eba5c2e723c11bb43775  guix-build-3df60704661c/output/arm-linux-gnueabihf/bitcoin-3df60704661c-arm-linux-gnueabihf.tar.gz
b836ca05dbc6521c15c07d4defebeb0da42d5e45add7eca22afd828178b97d5e  guix-build-3df60704661c/output/arm64-apple-darwin/SHA256SUMS.part
72f212a251d9ca25c3d0b0619fa405e92688aecf3b7fbf72ee510ca56f9f54f6  guix-build-3df60704661c/output/arm64-apple-darwin/bitcoin-3df60704661c-arm64-apple-darwin-unsigned.dmg
ec876babdbfc23c9381950ccbbe311f1e9106ff2efa58cd0821a0b5704e8586e  guix-build-3df60704661c/output/arm64-apple-darwin/bitcoin-3df60704661c-arm64-apple-darwin-unsigned.tar.gz
a359e6e1da3d3e9a3335a2c7f09ca4fa5b9ae5fa884dd3827d4dd29422b801bd  guix-build-3df60704661c/output/arm64-apple-darwin/bitcoin-3df60704661c-arm64-apple-darwin.tar.gz
5155153ca28a58c4c53cdd101567d3bb72f540a14f998a91d717ac1673b7ec6c  guix-build-3df60704661c/output/dist-archive/bitcoin-3df60704661c.tar.gz
afed3d1dc33acf2a7532eee12b090c379c205354ef21ee40d868187dab2d31ad  guix-build-3df60704661c/output/powerpc64-linux-gnu/SHA256SUMS.part
6914deb97cc52917fe59a80bf670ca171bcfca5a63fcf613b9072691a403a795  guix-build-3df60704661c/output/powerpc64-linux-gnu/bitcoin-3df60704661c-powerpc64-linux-gnu-debug.tar.gz
c8b6518fbed9181bd025a6e6f539fd16e5043e0aa887aec2a394448bacb250b2  guix-build-3df60704661c/output/powerpc64-linux-gnu/bitcoin-3df60704661c-powerpc64-linux-gnu.tar.gz
826feef8848521ecca935dd97edbf71b1616c83251c21fb667ceaac51e339cc2  guix-build-3df60704661c/output/powerpc64le-linux-gnu/SHA256SUMS.part
64c3a87cfe4b50c7b0561e208eb796ad46313d534ad6178e44280d5f2f78b757  guix-build-3df60704661c/output/powerpc64le-linux-gnu/bitcoin-3df60704661c-powerpc64le-linux-gnu-debug.tar.gz
668aa0c20fba5e23b15cf2971bd2235138ca333f7e5676a8ce270a30a9d3c01e  guix-build-3df60704661c/output/powerpc64le-linux-gnu/bitcoin-3df60704661c-powerpc64le-linux-gnu.tar.gz
05911bf9c6d3000bee0e8a57f181dfb9038ab957c93a0f5fa407610bfa012828  guix-build-3df60704661c/output/riscv64-linux-gnu/SHA256SUMS.part
b9be6c766869290fc18b10fe5dc7d5dd9c4e5e3d4ba75fcf3718595d74f69ce1  guix-build-3df60704661c/output/riscv64-linux-gnu/bitcoin-3df60704661c-riscv64-linux-gnu-debug.tar.gz
2f81847c316e7f5766cd6a50f87e9935f6257a4a22dfb15eabdf3c30f5599990  guix-build-3df60704661c/output/riscv64-linux-gnu/bitcoin-3df60704661c-riscv64-linux-gnu.tar.gz
0a4ae5b222ba181430a9ad7f9c5639d90fc40c94da8fe71da83dfff0f2119771  guix-build-3df60704661c/output/x86_64-apple-darwin/SHA256SUMS.part
c34705191796f2bbed9ee38a81866e68be076265a8215439fd29c0f501504c67  guix-build-3df60704661c/output/x86_64-apple-darwin/bitcoin-3df60704661c-x86_64-apple-darwin-unsigned.dmg
fd6c6dbcb6dd218d4463dc383ef107cd4b634ac21fd730e73f2d064981fb1ed3  guix-build-3df60704661c/output/x86_64-apple-darwin/bitcoin-3df60704661c-x86_64-apple-darwin-unsigned.tar.gz
8ff3c9b52bcb4b707dc6e6da7048f9ce9930810d4ca2c6b5e466ac956378d672  guix-build-3df60704661c/output/x86_64-apple-darwin/bitcoin-3df60704661c-x86_64-apple-darwin.tar.gz
b531a3269f27274a0a29a8ca5f4e9450d19d9c0e14e63bc2c078a0160e537488  guix-build-3df60704661c/output/x86_64-linux-gnu/SHA256SUMS.part
2926dc8f562ff0fc0944f5e27e68d86f0bc58c138bd20275b732b11911710714  guix-build-3df60704661c/output/x86_64-linux-gnu/bitcoin-3df60704661c-x86_64-linux-gnu-debug.tar.gz
69fa88d2d46cebbba109514c51ca694ec37a03c4c02e87a64f4b65dc4f7c6378  guix-build-3df60704661c/output/x86_64-linux-gnu/bitcoin-3df60704661c-x86_64-linux-gnu.tar.gz
801e5394b1561bf20461fa4494ea71ba02e61735b964e4393096f9077fc54a97  guix-build-3df60704661c/output/x86_64-w64-mingw32/SHA256SUMS.part
0d9b2b615630ad4d9a164dd05d316f386270261d4dc2ade9622bed999a0abb90  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64-debug.zip
a7f08c2d3ba4f93623b7a2634766bed1464523653f4223aa13302e3ce39b226a  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64-setup-unsigned.exe
855309534b379c3a27ecc9690d4553376a3aabb7383242bbe6f790ad43230bc5  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64-unsigned.tar.gz
d9c21190a3b1e437c9d2e25a2ee72e22848cbd00b6eb329e65f769c138c0ec7f  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64.zip
```

hebasto commented:
Guix builds:
```
109ed45706b8cbdc897b6c51efa0247f8e8f1e6bb35c57e69f30093b62189a51  guix-build-3df60704661c/output/aarch64-linux-gnu/SHA256SUMS.part
4e28463f1c074c7c121abd5fbad89aa2269c5978829ccd4898d33a21eedfeb4f  guix-build-3df60704661c/output/aarch64-linux-gnu/bitcoin-3df60704661c-aarch64-linux-gnu-debug.tar.gz
f20d074f0529202274a40f0530f79f601f5c575f1cd0addc099418719982485e  guix-build-3df60704661c/output/aarch64-linux-gnu/bitcoin-3df60704661c-aarch64-linux-gnu.tar.gz
414d7c4bf531a8d8589c15c9636ec0b092b28cba359b7c5ff46cd5bc0a768ac6  guix-build-3df60704661c/output/arm-linux-gnueabihf/SHA256SUMS.part
97e5cc1caf38e9e6cb4efde9be3a163c7373a1bdf1f7bdfbaa5fa7c9dd22a87c  guix-build-3df60704661c/output/arm-linux-gnueabihf/bitcoin-3df60704661c-arm-linux-gnueabihf-debug.tar.gz
cc174cb93c1b13b0efd5e002741ee6a13fcf1388f766eba5c2e723c11bb43775  guix-build-3df60704661c/output/arm-linux-gnueabihf/bitcoin-3df60704661c-arm-linux-gnueabihf.tar.gz
b836ca05dbc6521c15c07d4defebeb0da42d5e45add7eca22afd828178b97d5e  guix-build-3df60704661c/output/arm64-apple-darwin/SHA256SUMS.part
72f212a251d9ca25c3d0b0619fa405e92688aecf3b7fbf72ee510ca56f9f54f6  guix-build-3df60704661c/output/arm64-apple-darwin/bitcoin-3df60704661c-arm64-apple-darwin-unsigned.dmg
ec876babdbfc23c9381950ccbbe311f1e9106ff2efa58cd0821a0b5704e8586e  guix-build-3df60704661c/output/arm64-apple-darwin/bitcoin-3df60704661c-arm64-apple-darwin-unsigned.tar.gz
a359e6e1da3d3e9a3335a2c7f09ca4fa5b9ae5fa884dd3827d4dd29422b801bd  guix-build-3df60704661c/output/arm64-apple-darwin/bitcoin-3df60704661c-arm64-apple-darwin.tar.gz
5155153ca28a58c4c53cdd101567d3bb72f540a14f998a91d717ac1673b7ec6c  guix-build-3df60704661c/output/dist-archive/bitcoin-3df60704661c.tar.gz
afed3d1dc33acf2a7532eee12b090c379c205354ef21ee40d868187dab2d31ad  guix-build-3df60704661c/output/powerpc64-linux-gnu/SHA256SUMS.part
6914deb97cc52917fe59a80bf670ca171bcfca5a63fcf613b9072691a403a795  guix-build-3df60704661c/output/powerpc64-linux-gnu/bitcoin-3df60704661c-powerpc64-linux-gnu-debug.tar.gz
c8b6518fbed9181bd025a6e6f539fd16e5043e0aa887aec2a394448bacb250b2  guix-build-3df60704661c/output/powerpc64-linux-gnu/bitcoin-3df60704661c-powerpc64-linux-gnu.tar.gz
826feef8848521ecca935dd97edbf71b1616c83251c21fb667ceaac51e339cc2  guix-build-3df60704661c/output/powerpc64le-linux-gnu/SHA256SUMS.part
64c3a87cfe4b50c7b0561e208eb796ad46313d534ad6178e44280d5f2f78b757  guix-build-3df60704661c/output/powerpc64le-linux-gnu/bitcoin-3df60704661c-powerpc64le-linux-gnu-debug.tar.gz
668aa0c20fba5e23b15cf2971bd2235138ca333f7e5676a8ce270a30a9d3c01e  guix-build-3df60704661c/output/powerpc64le-linux-gnu/bitcoin-3df60704661c-powerpc64le-linux-gnu.tar.gz
05911bf9c6d3000bee0e8a57f181dfb9038ab957c93a0f5fa407610bfa012828  guix-build-3df60704661c/output/riscv64-linux-gnu/SHA256SUMS.part
b9be6c766869290fc18b10fe5dc7d5dd9c4e5e3d4ba75fcf3718595d74f69ce1  guix-build-3df60704661c/output/riscv64-linux-gnu/bitcoin-3df60704661c-riscv64-linux-gnu-debug.tar.gz
2f81847c316e7f5766cd6a50f87e9935f6257a4a22dfb15eabdf3c30f5599990  guix-build-3df60704661c/output/riscv64-linux-gnu/bitcoin-3df60704661c-riscv64-linux-gnu.tar.gz
0a4ae5b222ba181430a9ad7f9c5639d90fc40c94da8fe71da83dfff0f2119771  guix-build-3df60704661c/output/x86_64-apple-darwin/SHA256SUMS.part
c34705191796f2bbed9ee38a81866e68be076265a8215439fd29c0f501504c67  guix-build-3df60704661c/output/x86_64-apple-darwin/bitcoin-3df60704661c-x86_64-apple-darwin-unsigned.dmg
fd6c6dbcb6dd218d4463dc383ef107cd4b634ac21fd730e73f2d064981fb1ed3  guix-build-3df60704661c/output/x86_64-apple-darwin/bitcoin-3df60704661c-x86_64-apple-darwin-unsigned.tar.gz
8ff3c9b52bcb4b707dc6e6da7048f9ce9930810d4ca2c6b5e466ac956378d672  guix-build-3df60704661c/output/x86_64-apple-darwin/bitcoin-3df60704661c-x86_64-apple-darwin.tar.gz
b531a3269f27274a0a29a8ca5f4e9450d19d9c0e14e63bc2c078a0160e537488  guix-build-3df60704661c/output/x86_64-linux-gnu/SHA256SUMS.part
2926dc8f562ff0fc0944f5e27e68d86f0bc58c138bd20275b732b11911710714  guix-build-3df60704661c/output/x86_64-linux-gnu/bitcoin-3df60704661c-x86_64-linux-gnu-debug.tar.gz
69fa88d2d46cebbba109514c51ca694ec37a03c4c02e87a64f4b65dc4f7c6378  guix-build-3df60704661c/output/x86_64-linux-gnu/bitcoin-3df60704661c-x86_64-linux-gnu.tar.gz
801e5394b1561bf20461fa4494ea71ba02e61735b964e4393096f9077fc54a97  guix-build-3df60704661c/output/x86_64-w64-mingw32/SHA256SUMS.part
0d9b2b615630ad4d9a164dd05d316f386270261d4dc2ade9622bed999a0abb90  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64-debug.zip
a7f08c2d3ba4f93623b7a2634766bed1464523653f4223aa13302e3ce39b226a  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64-setup-unsigned.exe
855309534b379c3a27ecc9690d4553376a3aabb7383242bbe6f790ad43230bc5  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64-unsigned.tar.gz
d9c21190a3b1e437c9d2e25a2ee72e22848cbd00b6eb329e65f769c138c0ec7f  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64.zip
```

theuni commented:
> > just need to decide out what to do about guix.
> 
> What options are currently being considered. Is an upstream patch still possible?

FWIW, though we didn't end up needing it, an upstream guix "fix" is [slotted to go in as well](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=64100), which we can pick up in some future time-machine bump.

That gives us the ability to customize our clang build if we ever need to in the future.

fanquake reviewed (approved):
ACK 3df60704661cdb5e61ea2b999f468f3a1d16105f

hebasto commented:
> the runtime functionality required for `-Wl,-fixup_chains` [requires macOS >=11.0](https://github.com/llvm/llvm-project/blob/release/16.x/lld/MachO/Driver.cpp?rgh-link-date=2023-05-16T15%3A52%3A00Z#L1021).

However, [Xcode 13 Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-13-release-notes) state:
> All programs and dylibs built with a deployment target of macOS 12 or iOS 15 or later now use the chained fixups format. This uses different load commands and LINKEDIT data, and won’t run or load on older OS versions.

What effect will the `-Wl,-fixup_chains` option have for `OSX_MIN_VERSION=11.0`?

hebasto commented:
It would be nice if someone confirms that Guix binaries run on macOS 11 BigSur.

hebasto commented:
Guix (self-signed) `bitcoin-qt` tested on macOS Ventura 13.4.1 (Apple M1).

theuni commented:
> > the runtime functionality required for `-Wl,-fixup_chains` [requires macOS >=11.0](https://github.com/llvm/llvm-project/blob/release/16.x/lld/MachO/Driver.cpp?rgh-link-date=2023-05-16T15%3A52%3A00Z#L1021).
> 
> However, [Xcode 13 Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-13-release-notes) state:
> 
> > All programs and dylibs built with a deployment target of macOS 12 or iOS 15 or later now use the chained fixups format. This uses different load commands and LINKEDIT data, and won’t run or load on older OS versions.
> 
> What effect will the `-Wl,-fixup_chains` option have for `OSX_MIN_VERSION=11.0`?

> It would be nice if someone confirms that Guix binaries run on macOS 11 BigSur.

Yeah, that's a good point. The docs (and a quick glimpse at the code) shows that it should work fine on macOS 11, but as it wasn't default there it's worth confirming.

I'm also unable to test as I'm on 13.x :(

hebasto commented:
> Guix (self-signed) `bitcoin-qt` tested on macOS Ventura 13.4.1 (Apple M1).

Guix DMG tested on macOS Monterey 12.6.7 (Intel).

TheCharlatan commented:
I can confirm that my guix dmg runs on macOS 11.1.

sedited reviewed (approved):
ACK 3df60704661cdb5e61ea2b999f468f3a1d16105f

hebasto commented:
ACK 3df60704661cdb5e61ea2b999f468f3a1d16105f

tobtoht commented:
Guix builds:

```
109ed45706b8cbdc897b6c51efa0247f8e8f1e6bb35c57e69f30093b62189a51  guix-build-3df60704661c/output/aarch64-linux-gnu/SHA256SUMS.part
4e28463f1c074c7c121abd5fbad89aa2269c5978829ccd4898d33a21eedfeb4f  guix-build-3df60704661c/output/aarch64-linux-gnu/bitcoin-3df60704661c-aarch64-linux-gnu-debug.tar.gz
f20d074f0529202274a40f0530f79f601f5c575f1cd0addc099418719982485e  guix-build-3df60704661c/output/aarch64-linux-gnu/bitcoin-3df60704661c-aarch64-linux-gnu.tar.gz
414d7c4bf531a8d8589c15c9636ec0b092b28cba359b7c5ff46cd5bc0a768ac6  guix-build-3df60704661c/output/arm-linux-gnueabihf/SHA256SUMS.part
97e5cc1caf38e9e6cb4efde9be3a163c7373a1bdf1f7bdfbaa5fa7c9dd22a87c  guix-build-3df60704661c/output/arm-linux-gnueabihf/bitcoin-3df60704661c-arm-linux-gnueabihf-debug.tar.gz
cc174cb93c1b13b0efd5e002741ee6a13fcf1388f766eba5c2e723c11bb43775  guix-build-3df60704661c/output/arm-linux-gnueabihf/bitcoin-3df60704661c-arm-linux-gnueabihf.tar.gz
b836ca05dbc6521c15c07d4defebeb0da42d5e45add7eca22afd828178b97d5e  guix-build-3df60704661c/output/arm64-apple-darwin/SHA256SUMS.part
72f212a251d9ca25c3d0b0619fa405e92688aecf3b7fbf72ee510ca56f9f54f6  guix-build-3df60704661c/output/arm64-apple-darwin/bitcoin-3df60704661c-arm64-apple-darwin-unsigned.dmg
ec876babdbfc23c9381950ccbbe311f1e9106ff2efa58cd0821a0b5704e8586e  guix-build-3df60704661c/output/arm64-apple-darwin/bitcoin-3df60704661c-arm64-apple-darwin-unsigned.tar.gz
a359e6e1da3d3e9a3335a2c7f09ca4fa5b9ae5fa884dd3827d4dd29422b801bd  guix-build-3df60704661c/output/arm64-apple-darwin/bitcoin-3df60704661c-arm64-apple-darwin.tar.gz
5155153ca28a58c4c53cdd101567d3bb72f540a14f998a91d717ac1673b7ec6c  guix-build-3df60704661c/output/dist-archive/bitcoin-3df60704661c.tar.gz
afed3d1dc33acf2a7532eee12b090c379c205354ef21ee40d868187dab2d31ad  guix-build-3df60704661c/output/powerpc64-linux-gnu/SHA256SUMS.part
6914deb97cc52917fe59a80bf670ca171bcfca5a63fcf613b9072691a403a795  guix-build-3df60704661c/output/powerpc64-linux-gnu/bitcoin-3df60704661c-powerpc64-linux-gnu-debug.tar.gz
c8b6518fbed9181bd025a6e6f539fd16e5043e0aa887aec2a394448bacb250b2  guix-build-3df60704661c/output/powerpc64-linux-gnu/bitcoin-3df60704661c-powerpc64-linux-gnu.tar.gz
826feef8848521ecca935dd97edbf71b1616c83251c21fb667ceaac51e339cc2  guix-build-3df60704661c/output/powerpc64le-linux-gnu/SHA256SUMS.part
64c3a87cfe4b50c7b0561e208eb796ad46313d534ad6178e44280d5f2f78b757  guix-build-3df60704661c/output/powerpc64le-linux-gnu/bitcoin-3df60704661c-powerpc64le-linux-gnu-debug.tar.gz
668aa0c20fba5e23b15cf2971bd2235138ca333f7e5676a8ce270a30a9d3c01e  guix-build-3df60704661c/output/powerpc64le-linux-gnu/bitcoin-3df60704661c-powerpc64le-linux-gnu.tar.gz
05911bf9c6d3000bee0e8a57f181dfb9038ab957c93a0f5fa407610bfa012828  guix-build-3df60704661c/output/riscv64-linux-gnu/SHA256SUMS.part
b9be6c766869290fc18b10fe5dc7d5dd9c4e5e3d4ba75fcf3718595d74f69ce1  guix-build-3df60704661c/output/riscv64-linux-gnu/bitcoin-3df60704661c-riscv64-linux-gnu-debug.tar.gz
2f81847c316e7f5766cd6a50f87e9935f6257a4a22dfb15eabdf3c30f5599990  guix-build-3df60704661c/output/riscv64-linux-gnu/bitcoin-3df60704661c-riscv64-linux-gnu.tar.gz
0a4ae5b222ba181430a9ad7f9c5639d90fc40c94da8fe71da83dfff0f2119771  guix-build-3df60704661c/output/x86_64-apple-darwin/SHA256SUMS.part
c34705191796f2bbed9ee38a81866e68be076265a8215439fd29c0f501504c67  guix-build-3df60704661c/output/x86_64-apple-darwin/bitcoin-3df60704661c-x86_64-apple-darwin-unsigned.dmg
fd6c6dbcb6dd218d4463dc383ef107cd4b634ac21fd730e73f2d064981fb1ed3  guix-build-3df60704661c/output/x86_64-apple-darwin/bitcoin-3df60704661c-x86_64-apple-darwin-unsigned.tar.gz
8ff3c9b52bcb4b707dc6e6da7048f9ce9930810d4ca2c6b5e466ac956378d672  guix-build-3df60704661c/output/x86_64-apple-darwin/bitcoin-3df60704661c-x86_64-apple-darwin.tar.gz
b531a3269f27274a0a29a8ca5f4e9450d19d9c0e14e63bc2c078a0160e537488  guix-build-3df60704661c/output/x86_64-linux-gnu/SHA256SUMS.part
2926dc8f562ff0fc0944f5e27e68d86f0bc58c138bd20275b732b11911710714  guix-build-3df60704661c/output/x86_64-linux-gnu/bitcoin-3df60704661c-x86_64-linux-gnu-debug.tar.gz
69fa88d2d46cebbba109514c51ca694ec37a03c4c02e87a64f4b65dc4f7c6378  guix-build-3df60704661c/output/x86_64-linux-gnu/bitcoin-3df60704661c-x86_64-linux-gnu.tar.gz
801e5394b1561bf20461fa4494ea71ba02e61735b964e4393096f9077fc54a97  guix-build-3df60704661c/output/x86_64-w64-mingw32/SHA256SUMS.part
0d9b2b615630ad4d9a164dd05d316f386270261d4dc2ade9622bed999a0abb90  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64-debug.zip
a7f08c2d3ba4f93623b7a2634766bed1464523653f4223aa13302e3ce39b226a  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64-setup-unsigned.exe
855309534b379c3a27ecc9690d4553376a3aabb7383242bbe6f790ad43230bc5  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64-unsigned.tar.gz
d9c21190a3b1e437c9d2e25a2ee72e22848cbd00b6eb329e65f769c138c0ec7f  guix-build-3df60704661c/output/x86_64-w64-mingw32/bitcoin-3df60704661c-win64.zip
```

gruve-p commented:
ACK https://github.com/bitcoin/bitcoin/pull/27676/commits/3df60704661cdb5e61ea2b999f468f3a1d16105f
